packages/amplify-category-geo/src/provider-controllers/index.ts

```diff
@@ -0,0 +1,123 @@
+import { ProviderContext } from 'amplify-cli-core';
+import { ServiceName, provider } from '../service-utils/constants';
+import { $TSObject, open, stateManager } from 'amplify-cli-core';
+import { $TSContext } from 'amplify-cli-core';
+import { addPlaceIndexResource, updatePlaceIndexResource, removePlaceIndexResource } from './placeIndex';
+import { addMapResource, updateMapResource, removeMapResource } from './map';
+import { printer, prompter } from 'amplify-prompts';
+import { getServiceFriendlyName } from '../service-walkthroughs/resourceWalkthrough';
+
+/**
+ * Entry point for creating a new Geo resource
+ */
+export const addResource = async (
+  context: $TSContext,
+  service: string
+): Promise<string | undefined> => {
+  if(!projectHasAuth()) {
+    if (
+      await prompter.yesOrNo(
+        'geo category resources require auth (Amazon Cognito). Do you want to add auth now?',
+      )
+    ){
+      await context.amplify.invokePluginMethod(context, 'auth', undefined, 'add', [context]);
+    }
+    else {
+      printer.info('Please add auth (Amazon Cognito) to your project using "amplify add auth"');
```

Comment:
We are trying to avoid saying things like "please ..." 

```suggestion
      printer.info('Add auth (Amazon Cognito) to your project using "amplify add auth"');
```